MapboxNavigationUI/RouteMapViewController.swift

```diff
@@ -240,7 +240,46 @@ extension RouteMapViewController: NavigationMapViewDelegate {
             }
         }
         
-        return CLLocation(coordinate: newCoordinate, altitude: location.altitude, horizontalAccuracy: location.horizontalAccuracy, verticalAccuracy: location.verticalAccuracy, course: location.course, speed: location.speed, timestamp: location.timestamp)
+        let defaultReturn = CLLocation(coordinate: newCoordinate, altitude: location.altitude, horizontalAccuracy: location.horizontalAccuracy, verticalAccuracy: location.verticalAccuracy, course: location.course, speed: location.speed, timestamp: location.timestamp)
+        
+        guard location.course == -1 else {
+            return defaultReturn
+        }
+        
+        guard let coords = routeController.routeProgress.route.coordinates else {
+            return defaultReturn
+        }
+        
+        guard let closest = closestCoordinate(on: coords, to: location.coordinate) else {
+            return defaultReturn
+        }
+        let slicedLine = polyline(along: coords, from: closest.coordinate, to: coords.last)
+        
+        // Get closest point infront of user
+        guard let infrontPointOne = coordinate(at: 2, fromStartOf: slicedLine) else {
```

Comment:
Likewise, this method only returns `nil` if `polyline` is empty. We could make that clear in the function’s documentation.

(×2)